src/d/actor/d_a_ks.cpp

```diff
@@ -5,125 +5,1613 @@
+        }
+        case 0xc:
+            ks_kuttuki_check(i_this);
+
+            if (tyaku_check(i_this)) {
+                i_this->gravity = -4.0f;
+                i_this->speed.y = 1.0f;
+                i_this->speed.y += cM_rndF(5.0f);
+                i_this->speedF = 16.0f;
+            }
+        
+            if (i_this->m2E8[2] == 0) {
+                i_this->speedF = 0.0f;
+                i_this->m31C = 20.0f;
+                i_this->m2CB = 0;
+                i_this->m2CC = 0;
+            }
+    }
+
+    cLib_addCalcAngleS2(&i_this->shape_angle.y, i_this->current.angle.y, 1, 0x1000);
+    
+    if (body_atari_check(i_this)) {
+        fopAcM_seStart(i_this, JA_SE_LK_LAST_HIT, 0);
+    }
 }
 
 /* 00001630-00001874       .text action_kaze_move__FP8ks_class */
-void action_kaze_move(ks_class*) {
-    /* Nonmatching */
+void action_kaze_move(ks_class* i_this) {
+    switch (i_this->m2CC) {
+        case 0x14: {
+            for (int i = 0; i < 5; i++) {
+                i_this->m2F0[i] = 0;
+            }
+    
+            i_this->m2E8[1] = 0x32;
+    
+            fopAcM_monsSeStart(i_this, JA_SE_CV_KS_BLOW, 0);
+
+            i_this->m310 = i_this->current.pos.y;
+
+            i_this->m30C = 0.0f;
+
+            i_this->speedF = cM_rndF(15.0f) + 10.0f;
+            i_this->m314 = cM_rndF(15.0f) + 10.0f;
+
+            i_this->m308 = 0.0f;
+            
+            i_this->m2CC++;
+        }
+        case 0x15: {
+            cLib_addCalc2(&i_this->speed.y, i_this->m314, 0.8f, i_this->m308);
+            
+            cLib_addCalc2(&i_this->m308, 5.0f, 1.0f, 0.5f);
+            
+            if (i_this->m310 + 200.0f < i_this->current.pos.y || i_this->m2E8[1] == 0) {
+                i_this->m2CC++;
+            }
+
+            break;
+        }
+        case 0x16: {
+            cLib_addCalc2(&i_this->gravity, -1.0f, 0.3f, 0.5f);
+            
+            if (i_this->speed.y < -2.0f) {
+                i_this->speed.y = -2.0f;
+            }
+
+            cLib_addCalc0(&i_this->speedF, 0.5f, 2.0f);
+            
+            speed_keisan(i_this, i_this->shape_angle.y);
+
+            i_this->m2F0[0] += 3000;
+
+            i_this->m30C = 2.0f;
+
+            if (tyaku_check(i_this)) {
+                i_this->m2CB = 0;
+                i_this->m2CC = 0;
+                return;
+            }
+        }
+    }
+
+    i_this->shape_angle.z += 0x500;
+
+    if (body_atari_check(i_this)) {
+        fopAcM_seStart(i_this, JA_SE_LK_LAST_HIT, 0);
+    }
 }
 
+static u8 item_tbl[4] = {0xA, 0xB, 0xC, 0xA};
+
 /* 00001874-00001A14       .text dead_eff_set__FP8ks_classP4cXyz */
-void dead_eff_set(ks_class*, cXyz*) {
-    /* Nonmatching */
+void dead_eff_set(ks_class* i_this, cXyz* i_pos) {
+    fopAc_ac_c* a_this = (fopAc_ac_c*)i_this;
+    u8 health;
+    if (!strcmp(dComIfGp_getStartStageName(), "GanonK") && (i_this->stealItemBitNo != 0)) {
+        if (dComIfGs_getLife() <= 8) {
+            health = 10;
```

Comment:
```suggestion
            drop_type = daDisItem_HEART_e;
```
This "health" variable is actually what item should be dropped by the disappear, there's an enum for it: daDisappearItemType_e

I'll fix `fopAcM_createDisappear`'s param name to clarify this, "health" is an old name